frontend/pages/posts/[pageNumber].js

```diff
@@ -13,21 +12,26 @@ class Posts extends React.Component {
     pages: 10,
     maxPage: 100,
     loaded: false,
-    currentCollection: this.props.router.query.pageNumber
+    searchFilter: ''
   }
 
   componentDidMount() {
-    this.updatePostsDisplay()
+    //Forcing Asynchronous Order To Be Executed Last So Router Parameter is not Undefined
+    setTimeout(
+      () => this.updatePostsDisplay(this.props.router.query.pageNumber),
```

Comment:
This is a bit of a hack which I don't think would be an issue if you used `functional` components